lib/model/folder.go

```diff
@@ -39,6 +39,12 @@ type folder struct {
 	restartWatchChan chan struct{}
 	watchErr         error
 	watchErrMut      sync.Mutex
+
+	puller puller
```

Comment:
Still feels that sendRcv et all should embed folder, and then derived can just implement their own pull (and base folder panics).

Reply:
Yeah that's the intention. But embedded structs can't see their embedder to delegate functionality to them, so we need some hook in the other direction, either a func or interface value of some kind.

Reply:
Oh, that sucks, I though embedded.Foo() would call parent.Foo().